backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/NewCodeDefinition.java

```diff
@@ -62,6 +62,8 @@ static NewCodeDefinition withNumberOfDaysWithDate(int days, long thresholdDate)
     return new NewCodeNumberOfDaysWithDate(days, thresholdDate);
   }
 
+  long getThresholdDate();
```

Comment:
Why not returning an Instant from here? You can make the conversion later when you use it

Reply:
I think threshold date is maybe not precise enough. I am bad at naming, but what about something like "period start date"?

Reply:
Nitpick: I would move the method somewhere else, to not be in the middle of `with*` methods.

Reply:
Fixed the problem with Instant and long, now returning Instant.
Regarding the method location - I'm not sure where to move it, so I left it here for now.